homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -297,11 +308,13 @@ async def async_step_set_device_options(self, user_input=None):
             errors=errors,
         )
 
-    async def _async_replace_device(self, replace_device):
+    async def _async_replace_device(self, replace_device: str) -> None:
         """Migrate properties of a device into another."""
         device_registry = self._device_registry
         old_device = self._selected_device_entry_id
+        assert old_device
         old_entry = device_registry.async_get(old_device)
+        assert old_entry
```

Comment:
I think passing in `old_device` as an argument could simplify this so that the assertions are not needed here and in the caller -- just have it in one place, then lookup the device and pass it in to avoid having to keep track of when it's safe to call this function.  (I realize this is adding typing to existing code, but I think the asserts are pointing out complexity of the existing code)